### PR TITLE
Update .mailmap to get nicer 'git shortlog -sne' output

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
 Alexander Konovalov <alexander.konovalov@st-andrews.ac.uk> <alexk@mcs.st-andrews.ac.uk>
-Alexander Konovalov <alexander.konovalov@st-andrews.ac.uk> <alex-konovalov@users.noreply.github.com>
 Andreas Distler <adistler@fc.ul.pt> <andreas@mcs.st-and.ac.uk>
 Burkhard Höfling <burkhard@hoefling.name> <bh11@users.noreply.github.com>
 Chris Jefferson <caj21@st-andrews.ac.uk>
 Chris Jefferson <caj21@st-andrews.ac.uk> <chris@bubblescope.net>
 Chris Wensley <cdwensley.maths@btinternet.com> <cwensley1234@btinternet.com>
 Christopher Russell <cr66@st-andrews.ac.uk>
+Dmitrii Pasechnik <dimpase@gmail.com> <dima@pasechnik.info>
 Dmitrii Pasechnik <dimpase@gmail.com> <Dmitrii.Pasechnik@cs.ox.ac.uk>
```
Comment (fingolfin) [inline .mailmap]: @dimpase please let me know if `Dmitrii Pasechnik <dimpase@gmail.com>` is OK as your "preferred" git "identity" (or feel free to suggest something else)

File: .mailmap [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
-James Mitchell <jdm3@st-and.ac.uk> <jdm3@st-andrews.ac.uk>
+Gábor Horváth <ghorvath@science.unideb.hu> <hungaborhorvath@gmail.com>
 James Mitchell <jdm3@st-and.ac.uk> <james-d-mitchell@users.noreply.github.com>
+James Mitchell <jdm3@st-and.ac.uk> <jdm3@st-andrews.ac.uk>
 James Mitchell <jdm3@st-and.ac.uk> <jenkyfunds@gmail.com>
 James Williams <flowersinabox@hotmail.com>
 John McDermott <jjm@mcs.st-andrews.ac.uk> <jjm@mcs.st-and.ac.uk>
+Laurent Bartholdi <laurent.bartholdi@gmail.com>
+Lucas Wollenhaupt <github@wucas.de> <git@wucas.de>
+Lucas Wollenhaupt <github@wucas.de> <reg-github@wucas.de>
```
Comment (fingolfin) [inline .mailmap]: @wucas Please let me know if `Lucas Wollenhaupt <github@wucas.de>` is OK as your "preferred" git "identity" (or feel free to suggest something else)

File: .mailmap (hunk 1 above)
Comment (dimpase) [reply .mailmap]: Please use `Dmitrii Pasechnik <dima@pasechnik.info>`. Thanks for taking care of this.

File: .mailmap (hunk 2 above)
Comment (wucas) [reply .mailmap]: Yes that's alright, thanks!

**Pull request merged**